source/index.ts

```diff
@@ -184,6 +190,28 @@ class APIClient {
         return response.access_token;
     }
 
+    async _waitUploadSuccess(response: ItemResource, maxAwaitInProgressResponseSeconds: number): Promise<ItemResource> {
+        if (response.uploadState !== 'IN_PROGRESS') {
+            // We can return the response immediately if the upload is not in progress
+            return response;
+        }
+
+        const maxWait = maxAwaitInProgressResponseSeconds * 1000;
+        const retryInterval = 2000;
+
+        if (maxWait < 0) {
+            throw new Error('Upload is still in progress after maximum retries');
+        }
+
+        // Wait before checking again
+        await new Promise(resolve => {
+            setTimeout(resolve, retryInterval);
+        });
+
+        // Retry fetching the item resource
+        return this._waitUploadSuccess(await this.get('DRAFT'), maxAwaitInProgressResponseSeconds);
```

Comment:
`maxAwaitInProgressResponseSeconds` is being passed around unchanged, infinite retries

Reply:
Coffee didn't kick in yet 😶‍🌫️